### PR TITLE
Add .zap files used to generate the gen/ folders to examples/lighting…

### PR DESCRIPTION
**Pull request by vivien-apple**
…-app and examples/lock-app

 #### Problem
In order to makes it easier to move forward on #3464, I'm trying to extract the parts that are not fully dependent on ZAP.
It should makes it easier to understand the scope of various changes from #3464

This PR is the part of #3464 that includes the `.zap` file into the lock-app and lighting-app examples

 #### Summary of Changes
* Add `lock-app.zap` and `lighting-app.zap` to the common subfolders of the lock-app and lighting-app examples

### CODE REVIEW

File: examples/lighting-app/lighting-common/lighting-app.zap [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+      "key": "commandDiscovery",
+      "value": "1"
+    },
+    {
+      "key": "defaultResponsePolicy",
+      "value": "always"
+    },
+    {
+      "key": "filePath",
+      "value": "/Users/jepenven/workspace/connectedhomeip/examples/lock-app/lock-common/lock_app.zap"
```
Comment (andy31415) [inline examples/lighting-app/lighting-common/lighting-app.zap:16]: These paths seem to be user-specific. Are they ignored? Any chance we can make them generic somehow?

Comment (vivien-apple) [reply examples/lighting-app/lighting-common/lighting-app.zap:16]: The path seems to refer to itself. I'm not sure what is the uses for this key/value pair.

Adding @jepenven-silabs to understand what is the use of it.

File: examples/lighting-app/lighting-common/lighting-app.zap [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+    },
+    {
+      "key": "manufacturerCodes",
+      "value": "0x1002"
+    }
+  ],
+  "package": [
+    {
+      "path": "/Users/jepenven/workspace/zap/zcl-builtin/silabs/zcl.json",
```
Comment (rwalker-apple) [inline examples/lighting-app/lighting-common/lighting-app.zap:25]: these paths should be compiler arguments or workspace-relative.  this needs to be addressed in ZAP?

**Review by rwalker-apple — CHANGES_REQUESTED** [review]
absolute paths that refer to a specific user need to be addressed

**Review by mspang — CHANGES_REQUESTED** [review]
Mirroring the comments about paths.

File: examples/lighting-app/lighting-common/lighting-app.zap (hunk 1 above)
Comment (jepenven-silabs) [reply examples/lighting-app/lighting-common/lighting-app.zap:16]: It's a known issue with ZAP. Should be fixed by @tecimovic soon

Comment (tecimovic) [reply examples/lighting-app/lighting-common/lighting-app.zap:16]: We have on our plate to fix this shortly.
Appbuilder solution, which we plan to replicate, without having a different idea, was to store all paths in "relative to X" mode, where X could be:
   a.) relative to the ZAP file itself.
   b.) relative to the root of SDK (which is passed in as argument).
   c.) relative to the "project path", where IDE is responsible to pass in what "project path" is.
   d.) relative to the install location of zap
   e.) relative to the home directory of the user.
   f.) fully absolute.

User chooses in the UI what is the "relativity mode" for these paths, as they are stored in the zap file, and tooling enforces the existence of the proper "relativity point" in cases where it's required.

This worked for wide range of possibilities for collaborating teams which might decide how to standardize on stuff differently.

If anyone has a different good idea around this, I'll be all ears....

Comment (vivien-apple) [reply examples/lighting-app/lighting-common/lighting-app.zap:16]: > We have on our plate to fix this shortly.
> Appbuilder solution, which we plan to replicate, without having a different idea, was to store all paths in "relative to X" mode, where X could be:
> a.) relative to the ZAP file itself.
> b.) relative to the root of SDK (which is passed in as argument).
> c.) relative to the "project path", where IDE is responsible to pass in what "project path" is.
> d.) relative to the install location of zap
> e.) relative to the home directory of the user.
> f.) fully absolute.

Just curious. In the ZAP use case what is the difference between b.) and d.) currently ?

In our case a.) seems to fit well if I understand correctly.

**Comment by vivien-apple** [comment]
I have updated the PR with `.zap` files for the `lighting-app` and `lock-app`.

Those files does not contains absolute paths anymore, but relative paths to the ZAP folder.

The `ZAP` configuration file for the `lighting-app` is slightly different from the current configuration it uses. This is to take into account the fact that #3806 may land soon.

Similarly to others PR with `.zap` files you can generate parts of the `gen/` folders by running:
```
./scripts/tools/zap_generate.sh examples/lighting-app/lighting-common/lighting-app.zap
./scripts/tools/zap_generate.sh examples/lock-app/lock-common/lock-app.zap
```

**Review by mspang — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Pull request merged**